library/Vanilla/Formatting/Formats/HtmlFormat.php

```diff
@@ -225,4 +225,24 @@ protected function legacySpoilers(string $html): string {
         }
         return $html;
     }
+
+    /**
+     * Test whether a bit of HTML has legacy spoilers.
+     *
+     * @param string $html The HTML to test.
+     * @return bool
+     */
+    private function hasLegacySpoilers(string $html): bool {
+        // Check for an inline spoiler.
+        if (preg_match('`(\[spoiler\])[^\n]+(\[\/spoiler\])`', $html)) {
+            return true;
+        }
+
+        // Check for a multi-line spoiler.
+        if (preg_match_all('`^\[\/?spoiler\]$`m', $html, $m)) {
```

Comment:
Can this be simplified to a regular ole `preg_match`?

```suggestion
        if (preg_match('`^\[\/?spoiler\]$`m', $html)) {
```